### PR TITLE
Feature/33

### PR DESCRIPTION
**Pull request by uedamiku**
(2-3)商品の並び替え機能を追加しました。
レビューよろしくお願いします。

### CODE REVIEW

File: src/main/java/com/example/controller/ShowItemListController.java [hunk 1]
```diff
@@ -29,19 +31,28 @@ public class ShowItemListController {
 	 * 
 	 * @param model モデル
 	 * @param name  商品名
+	 * @param line  並び替え名
```
Comment (tkeshun) [inline src/main/java/com/example/controller/ShowItemListController.java:34]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

File: src/main/java/com/example/controller/ShowItemListController.java [hunk 2]
```diff
@@ -29,19 +31,28 @@ public class ShowItemListController {
...
 	 * @return 商品一覧画面
 	 */
 	@GetMapping("/show-item-list")
-	public String showItemList(Model model, String name) {
-		List<Item> itemList = showItemListService.searchByName(name);
+	public String showItemList(Model model, String name, String line) {
```
Comment (tkeshun) [inline src/main/java/com/example/controller/ShowItemListController.java:38]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

File: src/main/java/com/example/controller/ShowItemListController.java [hunk 3]
```diff
@@ -29,19 +31,28 @@ public class ShowItemListController {
...
+		List<Item> itemList = showItemListService.searchByName(name, line);
```
Comment (tkeshun) [inline src/main/java/com/example/controller/ShowItemListController.java:39]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

Comment (tkeshun) [inline src/main/java/com/example/controller/ShowItemListController.java:39]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.indentation.IndentationCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'method def' child has incorrect indentation level 16, expected level should be 4.

File: src/main/java/com/example/controller/ShowItemListController.java [hunk 4]
```diff
@@ -29,19 +31,28 @@ public class ShowItemListController {
...
 		if (itemList.size() == 0) {
-			itemList = showItemListService.showItemList();
+			itemList = showItemListService.showItemList(line);
```
Comment (tkeshun) [inline src/main/java/com/example/controller/ShowItemListController.java:41]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

Comment (tkeshun) [inline src/main/java/com/example/controller/ShowItemListController.java:41]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.indentation.IndentationCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'if' child has incorrect indentation level 24, expected level should be 6.

File: src/main/java/com/example/controller/ShowItemListController.java [hunk 5]
```diff
@@ -29,19 +31,28 @@ public class ShowItemListController {
...
 			model.addAttribute("message", "該当する商品がありません");
 		}
 		if (itemList.size() == 0) {
 			model.addAttribute("message", "商品は1件も存在しません");
 		}
+
+		Map<String, String> lineMap = new HashMap<>();
```
Comment (tkeshun) [inline src/main/java/com/example/controller/ShowItemListController.java:48]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

Comment (tkeshun) [inline src/main/java/com/example/controller/ShowItemListController.java:48]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.indentation.IndentationCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'method def' child has incorrect indentation level 16, expected level should be 4.

File: src/main/java/com/example/controller/ShowItemListController.java [hunk 6]
```diff
@@ -29,19 +31,28 @@ public class ShowItemListController {
...
+		lineMap.put("initi", "名前順");
```
Comment (tkeshun) [inline src/main/java/com/example/controller/ShowItemListController.java:49]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

Comment (tkeshun) [inline src/main/java/com/example/controller/ShowItemListController.java:49]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.indentation.IndentationCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'method def' child has incorrect indentation level 16, expected level should be 4.

File: src/main/java/com/example/controller/ShowItemListController.java [hunk 7]
```diff
@@ -29,19 +31,28 @@ public class ShowItemListController {
...
+		lineMap.put("low", "価格の安い順");
```
Comment (tkeshun) [inline src/main/java/com/example/controller/ShowItemListController.java:50]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

Comment (tkeshun) [inline src/main/java/com/example/controller/ShowItemListController.java:50]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.indentation.IndentationCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'method def' child has incorrect indentation level 16, expected level should be 4.

File: src/main/java/com/example/controller/ShowItemListController.java [hunk 8]
```diff
@@ -29,19 +31,28 @@ public class ShowItemListController {
...
+		lineMap.put("high", "価格の高い順");
```
Comment (tkeshun) [inline src/main/java/com/example/controller/ShowItemListController.java:51]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

Comment (tkeshun) [inline src/main/java/com/example/controller/ShowItemListController.java:51]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.indentation.IndentationCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'method def' child has incorrect indentation level 16, expected level should be 4.

File: src/main/java/com/example/controller/ShowItemListController.java [hunk 9]
```diff
@@ -29,19 +31,28 @@ public class ShowItemListController {
...
+		model.addAttribute("lineMap", lineMap);
```
Comment (tkeshun) [inline src/main/java/com/example/controller/ShowItemListController.java:52]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

Comment (tkeshun) [inline src/main/java/com/example/controller/ShowItemListController.java:52]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.indentation.IndentationCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'method def' child has incorrect indentation level 16, expected level should be 4.

File: src/main/java/com/example/controller/ShowItemListController.java [hunk 10]
```diff
@@ -29,19 +31,28 @@ public class ShowItemListController {
...
+
 		model.addAttribute("itemList", itemList);
+		model.addAttribute("linename", line);
```
Comment (tkeshun) [inline src/main/java/com/example/controller/ShowItemListController.java:55]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

Comment (tkeshun) [inline src/main/java/com/example/controller/ShowItemListController.java:55]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.indentation.IndentationCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'method def' child has incorrect indentation level 16, expected level should be 4.

File: src/main/java/com/example/repository/ItemRepository.java [hunk 11]
```diff
@@ -30,13 +30,22 @@ public class ItemRepository {
 	private NamedParameterJdbcTemplate template;
 
 	/**
-	 * 商品一覧情報を値段の安い順で取得します.
+	 * 商品一覧情報を任意の順で取得します.(価格の安い順、高い順、名前の昇順)
```
Comment (tkeshun) [inline src/main/java/com/example/repository/ItemRepository.java:33]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

File: src/main/java/com/example/repository/ItemRepository.java [hunk 12]
```diff
@@ -30,13 +30,22 @@ public class ItemRepository {
...
 	 * 
+	 * @param line 並び替え名
```
Comment (tkeshun) [inline src/main/java/com/example/repository/ItemRepository.java:35]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.javadoc.RequireEmptyLineBeforeBlockTagGroupCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Javadoc tag '@param' should be preceded with an empty line.

Comment (tkeshun) [inline src/main/java/com/example/repository/ItemRepository.java:35]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

File: src/main/java/com/example/repository/ItemRepository.java [hunk 13]
```diff
@@ -30,13 +30,22 @@ public class ItemRepository {
...
 	 * @return 全商品一覧(商品が存在しない場合はサイズ0件の商品一覧を返します)
 	 */
-	public List<Item> findAll() {
-		String sql = "SELECT id,name,description,price_m,price_l,image_path,deleted FROM items ORDER BY price_m,id;";
-		List<Item> itemList = template.query(sql, ITEM_ROW_MAPPER);
+	public List<Item> findAll(String line) {
```
Comment (tkeshun) [inline src/main/java/com/example/repository/ItemRepository.java:38]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

Comment (tkeshun) [inline src/main/java/com/example/repository/ItemRepository.java:38]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.indentation.IndentationCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'method def modifier' has incorrect indentation level 8, expected level should be 2.

File: src/main/java/com/example/repository/ItemRepository.java [hunk 14]
```diff
@@ -30,13 +30,22 @@ public class ItemRepository {
...
+		StringBuilder sql = new StringBuilder();
```
Comment (tkeshun) [inline src/main/java/com/example/repository/ItemRepository.java:39]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

Comment (tkeshun) [inline src/main/java/com/example/repository/ItemRepository.java:39]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.indentation.IndentationCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'method def' child has incorrect indentation level 16, expected level should be 4.

File: src/main/java/com/example/repository/ItemRepository.java [hunk 15]
```diff
@@ -30,13 +30,22 @@ public class ItemRepository {
...
+		sql.append("SELECT id,name,description,price_m,price_l,image_path,deleted FROM items ORDER BY ");
```
Comment (tkeshun) [inline src/main/java/com/example/repository/ItemRepository.java:40]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.sizes.LineLengthCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line is longer than 100 characters (found 113).

Comment (tkeshun) [inline src/main/java/com/example/repository/ItemRepository.java:40]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

Comment (tkeshun) [inline src/main/java/com/example/repository/ItemRepository.java:40]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.indentation.IndentationCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'method def' child has incorrect indentation level 16, expected level should be 4.

File: src/main/java/com/example/repository/ItemRepository.java [hunk 16]
```diff
@@ -30,13 +30,22 @@ public class ItemRepository {
...
+		if ("low".equals(line)) {
```
Comment (tkeshun) [inline src/main/java/com/example/repository/ItemRepository.java:41]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

Comment (tkeshun) [inline src/main/java/com/example/repository/ItemRepository.java:41]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.indentation.IndentationCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'if' has incorrect indentation level 16, expected level should be 4.

**Review by tkeshun — COMMENTED** [review]
Remaining comments which cannot be posted as a review comment to avoid GitHub Rate Limit

<details>
<summary>checkstyle</summary>

[src/main/java/com/example/repository/ItemRepository.java|42 col 1|](http://github.com/ecsite-project/ec-202304c/blob/0c23f70796b1b9046edc4d7f0fcb567276668510/src/main/java/com/example/repository/ItemRepository.java#L42) Line contains a tab character.
[src/main/java/com/example/repository/ItemRepository.java|42 col 25|](http://github.com/ecsite-project/ec-202304c/blob/0c23f70796b1b9046edc4d7f0fcb567276668510/src/main/java/com/example/repository/ItemRepository.java#L42) 'if' child has incorrect indentation level 24, expected level should be 6.
[src/main/java/com/example/repository/ItemRepository.java|43 col 1|](http://github.com/ecsite-project/ec-202304c/blob/0c23f70796b1b9046edc4d7f0fcb567276668510/src/main/java/com/example/repository/ItemRepository.java#L43) Line contains a tab character.
[src/main/java/com/example/repository/ItemRepository.java|43 col 17|](http://github.com/ecsite-project/ec-202304c/blob/0c23f70796b1b9046edc4d7f0fcb567276668510/src/main/java/com/example/repository/ItemRepository.java#L43) 'if rcurly' has incorrect indentation level 16, expected level should be 4.
[src/main/java/com/example/repository/ItemRepository.java|44 col 1|](http://github.com/ecsite-project/ec-202304c/blob/0c23f70796b1b9046edc4d7f0fcb567276668510/src/main/java/com/example/repository/ItemRepository.java#L44) Line contains a tab character.
[src/main/java/com/example/repository/ItemRepository.java|44 col 25|](http://github.com/ecsite-project/ec-202304c/blob/0c23f70796b1b9046edc4d7f0fcb567276668510/src/main/java/com/example/repository/ItemRepository.java#L44) 'if' child has incorrect indentation level 24, expected level should be 6.
[src/main/java/com/example/repository/ItemRepository.java|45 col 1|](http://github.com/ecsite-project/ec-202304c/blob/0c23f70796b1b9046edc4d7f0fcb567276668510/src/main/java/com/example/repository/ItemRepository.java#L45) Line contains a tab character.
[src/main/java/com/example/repository/ItemRepository.java|45 col 17|](http://github.com/ecsite-project/ec-202304c/blob/0c23f70796b1b9046edc4d7f0fcb567276668510/src/main/java/com/example/repository/ItemRepository.java#L45) 'if rcurly' has incorrect indentation level 16, expected level should be 4.
[src/main/java/com/example/repository/ItemRepository.java|46 col 1|](http://github.com/ecsite-project/ec-202304c/blob/0c23f70796b1b9046edc4d7f0fcb567276668510/src/main/java/com/example/repository/ItemRepository.java#L46) Line contains a tab character.
[src/main/java/com/example/repository/ItemRepository.java|46 col 25|](http://github.com/ecsite-project/ec-202304c/blob/0c23f70796b1b9046edc4d7f0fcb567276668510/src/main/java/com/example/repository/ItemRepository.java#L46) 'if' child has incorrect indentation level 24, expected level should be 6.
[src/main/java/com/example/repository/ItemRepository.java|47 col 1|](http://github.com/ecsite-project/ec-202304c/blob/0c23f70796b1b9046edc4d7f0fcb567276668510/src/main/java/com/example/repository/ItemRepository.java#L47) Line contains a tab character.
[src/main/java/com/example/repository/ItemRepository.java|47 col 17|](http://github.com/ecsite-project/ec-202304c/blob/0c23f70796b1b9046edc4d7f0fcb567276668510/src/main/java/com/example/repository/ItemRepository.java#L47) 'if rcurly' has incorrect indentation level 16, expected level should be 4.
[src/main/java/com/example/repository/ItemRepository.java|48 col 1|](http://github.com/ecsite-project/ec-202304c/blob/0c23f70796b1b9046edc4d7f0fcb567276668510/src/main/java/com/example/repository/ItemRepository.java#L48) Line contains a tab character.
[src/main/java/com/example/repository/ItemRepository.java|48 col 17|](http://github.com/ecsite-project/ec-202304c/blob/0c23f70796b1b9046edc4d7f0fcb567276668510/src/main/java/com/example/repository/ItemRepository.java#L48) 'method def' child has incorrect indentation level 16, expected level should be 4.
[src/main/java/com/example/service/ShowItemListService.java|27|](http://github.com/ecsite-project/ec-202304c/blob/0c23f70796b1b9046edc4d7f0fcb567276668510/src/main/java/com/example/service/ShowItemListService.java#L27) Javadoc tag '@param' should be preceded with an empty line.
[src/main/java/com/example/service/ShowItemListService.java|27 col 1|](http://github.com/ecsite-project/ec-202304c/blob/0c23f70796b1b9046edc4d7f0fcb567276668510/src/main/java/com/example/service/ShowItemListService.java#L27) Line contains a tab character.
[src/main/java/com/example/service/ShowItemListService.java|30 col 1|](http://github.com/ecsite-project/ec-202304c/blob/0c23f70796b1b9046edc4d7f0fcb567276668510/src/main/java/com/example/service/ShowItemListService.java#L30) Line contains a tab character.
[src/main/java/com/example/service/ShowItemListService.java|30 col 9|](http://github.com/ecsite-project/ec-202304c/blob/0c23f70796b1b9046edc4d7f0fcb567276668510/src/main/java/com/example/service/ShowItemListService.java#L30) 'method def modifier' has incorrect indentation level 8, expected level should be 2.
[src/main/java/com/example/service/ShowItemListService.java|31 col 1|](http://github.com/ecsite-project/ec-202304c/blob/0c23f70796b1b9046edc4d7f0fcb567276668510/src/main/java/com/example/service/ShowItemListService.java#L31) Line contains a tab character.
[src/main/java/com/example/service/ShowItemListService.java|31 col 17|](http://github.com/ecsite-project/ec-202304c/blob/0c23f70796b1b9046edc4d7f0fcb567276668510/src/main/java/com/example/service/ShowItemListService.java#L31) 'method def' child has incorrect indentation level 16, expected level should be 4.
[src/main/java/com/example/service/ShowItemListService.java|42 col 1|](http://github.com/ecsite-project/ec-202304c/blob/0c23f70796b1b9046edc4d7f0fcb567276668510/src/main/java/com/example/service/ShowItemListService.java#L42) Line contains a tab character.
[src/main/java/com/example/service/ShowItemListService.java|42 col 9|](http://github.com/ecsite-project/ec-202304c/blob/0c23f70796b1b9046edc4d7f0fcb567276668510/src/main/java/com/example/service/ShowItemListService.java#L42) 'method def modifier' has incorrect indentation level 8, expected level should be 2.
[src/main/java/com/example/service/ShowItemListService.java|44 col 1|](http://github.com/ecsite-project/ec-202304c/blob/0c23f70796b1b9046edc4d7f0fcb567276668510/src/main/java/com/example/service/ShowItemListService.java#L44) Line contains a tab character.
[src/main/java/com/example/service/ShowItemListService.java|44 col 25|](http://github.com/ecsite-project/ec-202304c/blob/0c23f70796b1b9046edc4d7f0fcb567276668510/src/main/java/com/example/service/ShowItemListService.java#L44) 'if' child has incorrect indentation level 24, expected level should be 6.
</details>

**Review by yousuke-murayama — APPROVED** [review]
問題ないと思います。

**Pull request merged**